incubator/hnc/internal/pkg/selectors/selectors.go

```diff
@@ -0,0 +1,70 @@
+// Package selectors contains exceptions related utilities, including check the validity of selector,
+// treeSelector, and noneSelector, and parsing these three selectors.
+package selectors
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/go-logr/logr"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+
+	api "sigs.k8s.io/multi-tenancy/incubator/hnc/api/v1alpha2"
+)
+
+func GetSelectorAnnotation(inst *unstructured.Unstructured) (string, bool) {
+	annot := inst.GetAnnotations()
+	str, ok := annot[api.AnnotationSelector]
+	return str, ok
+}
+
+// GetTreeSelector is similar to a regular selector, except that it adds the LabelTreeDepthSuffix to every string
+// To transform a tree selector into a regular label selector, we follow these steps:
+// 1. get the treeSelector annotation if it exists
+// 2. convert the annotation string to a slice of strings seperated by comma, because user is allowed to put multiple selectors
+// 3. append the LabelTreeDepthSuffix to each of the treeSelector string
+// 4. combine them into a single string connected by comma
+func GetTreeSelector(log logr.Logger, inst *unstructured.Unstructured) (labels.Selector, error) {
+	annot := inst.GetAnnotations()
+	treeSelectorStr, ok := annot[api.AnnotationTreeSelector]
+	if !ok {
+		return nil, errors.New("TreeSelector not found.")
```

Comment:
It's not an error condition for an object to _not_ have a tree selector - in fact, that's the normal case. So we shouldn't return an error here IMO, just `nil, nil`.

Reply:
done